spec/takeaway_spec.rb

```diff
@@ -0,0 +1,71 @@
+require 'takeaway'
+
+describe Takeaway do
+let(:menu) { double :menu }
+let(:first_order) {double :first_order }
+subject {described_class.new(menu)}
+
+
+it { is_expected.to respond_to(:pick).with(2).arguments }
+  
+it { is_expected.to respond_to(:check_sum) }
+
+it { is_expected.to respond_to(:basket) }
+
+  describe '#initialize' do
+    it 'creates a menu' do
+    expect(subject.menu).to eq menu
+    end
+  end
+
+ describe '#new_order' do
+   
+   it 'creates a new order' do
+    allow(subject).to receive(:new_order).and_return first_order
```

Comment:
Never stub methods on the subject. You only want to mock external dependencies (other classes, the system, etc).

Reply:
Here, your `new_order` method is set-up for dependency injection already, so you can easily do something like 
```
subject.new_order(first_order)
```
to achieve the same thing.